Sources/TuistKit/Services/FetchService.swift

```diff
@@ -14,34 +14,65 @@ final class FetchService {
     private let dependenciesController: DependenciesControlling
     private let dependenciesModelLoader: DependenciesModelLoading
     private let converter: ManifestModelConverting
+    private let rootDirectoryLocator: RootDirectoryLocating
+    private let fileHandler: FileHandling
 
     init(
         pluginService: PluginServicing = PluginService(),
         configLoader: ConfigLoading = ConfigLoader(manifestLoader: CachedManifestLoader()),
         manifestLoader: ManifestLoading = ManifestLoader(),
         dependenciesController: DependenciesControlling = DependenciesController(),
         dependenciesModelLoader: DependenciesModelLoading = DependenciesModelLoader(),
-        converter: ManifestModelConverting = ManifestModelConverter()
+        converter: ManifestModelConverting = ManifestModelConverter(),
+        rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator(),
+        fileHandler: FileHandling = FileHandler.shared
     ) {
         self.pluginService = pluginService
         self.configLoader = configLoader
         self.manifestLoader = manifestLoader
         self.dependenciesController = dependenciesController
         self.dependenciesModelLoader = dependenciesModelLoader
         self.converter = converter
+        self.rootDirectoryLocator = rootDirectoryLocator
+        self.fileHandler = fileHandler
     }
 
     func run(
         path: String?,
         update: Bool
     ) async throws {
-        let path = try self.path(path)
-
+        let path = try locateDependencies(at: path)
         try await fetchDependencies(path: path, update: update, with: fetchPlugins(path: path))
     }
 
     // MARK: - Helpers
 
+    public func locateDependencies(at path: String?) throws -> AbsolutePath {
+        // Convert to AbsolutePath
+        let path = try self.path(path)
+
+        // If the Dependencies.swift file exists in the root Tuist directory, we load it from there
+        if let rootDirectoryPath = rootDirectoryLocator.locate(from: path) {
+            if fileHandler.exists(
+                rootDirectoryPath.appending(components: Constants.tuistDirectoryName, Manifest.dependencies.fileName(path))
+            ) {
+                return rootDirectoryPath
+            }
+        }
+
+        // Else, we try to traverse up the directories to find it
+        if let traversedPath = fileHandler.locateDirectoryTraversingParents(from: path, path: Constants.tuistDirectoryName) {
+            if fileHandler.exists(
+                traversedPath.appending(components: Constants.tuistDirectoryName, Manifest.dependencies.fileName(path))
+            ) {
+                return traversedPath
+            }
+        }
```

Comment:
What is this for? I don't think we support intermediate Tuist folders, so it should be enough to always return the root

Reply:
I saw we support it with the [Config.swift ](https://github.com/tuist/tuist/blob/main/Sources/TuistLoader/Loaders/ConfigLoader.swift#L66) - I can remove if needed

Reply:
I agree with @danyf90. Let's remove that piece of code.

Reply:
Fixed